pkg/services/sqlstore/migrations/ualert/channel.go

```diff
@@ -115,19 +114,20 @@ func (m *migration) setupAlertmanagerConfigs(rulesPerOrg map[int64]map[*alertRul
 	return amConfigPerOrg, nil
 }
 
-func keysString(m map[string]interface{}) (string, error) {
+// contactListToString creates a sorted string representation of a given map (set) of receiver names. Each name will be comma-separated and double-quoted. Double quotes inside a name are escaped.
+func contactListToString(m map[string]interface{}) string {
 	keys := make([]string, 0, len(m))
 	for k := range m {
-		keys = append(keys, k)
+		keys = append(keys, quote(k))
 	}
 	sort.Strings(keys)
 
-	jsonStr, err := json.Marshal(keys)
-	if err != nil {
-		return "", err
-	}
+	return strings.Join(keys, ",")
+}
 
-	return string(jsonStr), nil
+// quote will surround the given string in double quotes and escape any existing interior double quotes.
+func quote(s string) string {
+	return `"` + strings.ReplaceAll(s, `"`, `\"`) + `"`
```

Comment:
You might need to escape existing escaped quotes?

Reply:
I'm not 100% sure on this, but if we have `\"` then we will get `\\"` when what we want I think is `\\\"`.

Reply:
You're right but extra escaping won't help either. Now that you point this out, I realize that my original intention for escaping quotes here doesn't accomplish what I had hoped.

Basically, I was attempting to use the outside quotes to separate contact names to make sure that partial names can't accidentally be matched by the policy regex `.*"<name>".*`. So, I escaped interior quotes so they wouldn't match the boundary. This doesn't work though, for example we could have two (rather evil) contact points:

- `hello\`
- `hello"world`

The regex for `hello\` would be `.*"hello\\".*` which would also incorrectly match `hello"world`'s contact label which would be `"hello\"world"`. [regex101 demonstration](https://regex101.com/r/3xOxye/1).

I'm sure there's a way to make the regex more robust using some lookahead or alternation but it won't be pretty... might be "good enough" in this state. Alternatively we can go the non-regex route and create multiple labels on each alert rule (one per contact).

Reply:
I think that the regex method is too brittle and too much hassle for the benefit. I'm going to push a change that uses multiple labels shortly and edit the description/screenshots.

Reply:
After some internal discussion, we decided to go back to the regex approach but sanitize any double quotes internal to the contact name on migration and replace them with `_`